src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+[`StatelessWidget`](https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html).
+
+However, if the unique characteristics of a widget need to change based on user
+interaction or other factors, that widget is _stateful_. For example, if a
+widget has a counter that increments whenever the user taps a button, then the
+value of the counter is the state for that widget. When that value changes, the
+widget needs to be rebuilt to update its part of the UI. These widgets subclass
+[`StatefulWidget`](https://api.flutter.dev/flutter/widgets/StatefulWidget-class.html), and (because the widget itself is immutable) they store mutable state in a separate class that subclasses [`State`](https://api.flutter.dev/flutter/widgets/State-class.html). `StatefulWidget`s don’t have a build method; instead, their user interface is built through their `State` object.
+
+Whenever you mutate a `State` object (for example, by incrementing the counter),
+you must call
+[`setState()`](https://api.flutter.dev/flutter/widgets/State/setState.html) to signal the framework to update the user interface by calling the `State`’s build method again.
+
+Having separate state and widget objects lets other widgets treat both stateless
+and stateful widgets in exactly the same way, without being concerned about
+losing state. Instead of needing to hold on to a child to preserve its state,
+the parent can create a new instance of the child at any time without losing the
+child’s persistent state. The framework does all the work of finding and reusing
+existing state objects when appropriate.
+
+### State management
+
+So, if many widgets can contain state, how is state managed and passed around
+the system?
+
+As with any other class, you can use a constructor in a widget to initialize its
+data, so a `build()` method can ensure that any child widget is instantiated
+with the data it needs:
+
+```dart
+@override
+Widget build(BuildContext context) {
+   return ContentWidget(importantState);
+}
+```
+
+As widget trees get deeper, however, passing state information up and down the
+tree hierarchy becomes cumbersome. So, a third widget type,
+[`InheritedWidget`](https://api.flutter.dev/flutter/widgets/InheritedWidget-class.html), provides an easy way to grab data from a shared ancestor. You can use `InheritedWidget` to create a state widget that wraps a common
+ancestor in the widget tree, as shown in this example:
+
+![Inherited
+widgets](../../images/arch-overview/inherited-widget.svg){:width="100%"}
+
+Whenever one of the `ExamWidget` or `GradeWidget` objects needs data from
+`StudentState`, it can now access it with a command such as:
+
+```dart
+final studentState = StudentState.of(context);
+```
+
+The `of(context)` call takes the build context (a handle to the current widget
+location), and returns [the nearest ancestor in the
+tree](https://master-api.flutter.dev/flutter/widgets/BuildContext/dependOnInheritedWidgetOfExactType.html)
```

Comment:
Ooh. Must we link to the master API? Can you link to the regular API, not the master? (I see this is available in both.) Switching to the {{site.api}} macro will take care of that.

Reply:
Absolutely! Fixed.